resize-observer/contain-instrinsic-size-should-not-leak-nodes.html increase gc attempts

stwrt opened the pull request:
#### a0afb04dda939844efedf013d8370de633306c37
<pre>
resize-observer/contain-instrinsic-size-should-not-leak-nodes.html increase gc attempts
<a href="https://bugs.webkit.org/show_bug.cgi?id=292686">https://bugs.webkit.org/show_bug.cgi?id=292686</a>
<a href="https://rdar.apple.com/150302301">rdar://150302301</a>

Reviewed by Ryan Reno.

We need to increase the amount of attempts we do gc() as it is non-deterministic if they will
all be collected on the first pass.

* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-site-isolation/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/platform/visionos/TestExpectations:
* LayoutTests/platform/win/TestExpectations:
* LayoutTests/resize-observer/contain-instrinsic-size-should-not-leak-nodes-expected.txt:
* LayoutTests/resize-observer/contain-instrinsic-size-should-not-leak-nodes.html:

Canonical link: <a href="https://commits.webkit.org/294696@main">https://commits.webkit.org/294696@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9e93ed7ad01b0cd283bfac47a3c094a3695040dd

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/102676 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/22349 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/12667 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/107841 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/53317 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/22709 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/30851 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/78092 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/35065 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/105682 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/17591 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/92668 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/58424 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/17432 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/10700 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/52674 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/87248 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10770 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/110217 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/29813 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/21982 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/87074 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/30177 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88860 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86679 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22078 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/31509 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/9242 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/24066 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/29740 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/29548 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/32875 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/31110 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rreno commented on LayoutTests/resize-observer/contain-instrinsic-size-should-not-leak-nodes.html:
```diff
@@ -77,7 +93,23 @@
-    assert_equals(internals.referencingNodeCount(document) < initialNodeCount + CISCount * 0.8, true, 'More than 20% of nodes should be collected');
+
+    let checkCount = 0;
+    let handle = setInterval(() => {
```
It's a little awkward to have this in the middle of a promise test. I'd recommend wrapping this interval up in a function returning a promise that can be awaited here.

Then you can either `resolve` or `reject` from within the `setInterval` depending on how the test played out.

rreno reviewed (approved):
Thanks! 😄

stwrt committed:
resize-observer/contain-instrinsic-size-should-not-leak-nodes.html increase gc attempts
https://bugs.webkit.org/show_bug.cgi?id=292686
rdar://150302301

Reviewed by Ryan Reno.

We need to increase the amount of attempts we do gc() as it is non-deterministic if they will
all be collected on the first pass.

* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-site-isolation/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/platform/visionos/TestExpectations:
* LayoutTests/platform/win/TestExpectations:
* LayoutTests/resize-observer/contain-instrinsic-size-should-not-leak-nodes-expected.txt:
* LayoutTests/resize-observer/contain-instrinsic-size-should-not-leak-nodes.html:

Canonical link: https://commits.webkit.org/294696@main
